### PR TITLE
Fixed the problem that caused installation to fail.

### LINKED ISSUE #126: Cannot install nnmnkwii because of cythonizing error
I failed to install nnmnkwii which is the HEAD version of master branch  because cnp.int_t in nnmnkwii/paramgan/_bandmat/misc.pyx is invalid type when cythonizing.  I'm afraid that it came from the API incompatibility between numpy 2.0.0 and that of older versions.  

environment: Debian 12.5(WSL2)
python version: 3.11.2
numpy version: 2.0.0
cython version: 3.0.10

[nnmnkwii_install.log](https://github.com/user-attachments/files/15911596/nnmnkwii_install.log)

### PR DESCRIPTION
**Pull request by decfrr**
## What
Fix #126 

## Why
The problem was due to an unset version limit for numpy.
With [the release of version 2.0.0](https://pypi.org/project/numpy/2.0.0/) on the 16th of this month, which contained some destructive changes, the build of this package was failing.
By setting the upper version limit, the build will succeed.

## Tasks
- [x] Passed all tests
- [x] Successful build

```
Successfully built nnmnkwii pysptk fastdtw
Installing collected packages: tqdm, threadpoolctl, scipy, joblib, fastdtw, decorator, cython, scikit-learn, pysptk, nnmnkwii
Successfully installed cython-3.0.10 decorator-5.1.1 fastdtw-0.3.4 joblib-1.4.2 nnmnkwii-0.1.2+67ff5ff pysptk-0.2.2 scikit-learn-1.5.0 scipy-1.13.1 threadpoolctl-3.5.0 tqdm-4.66.4
```

### CODE REVIEW

**Comment by r9y9** [comment]
Looks good to me. Can you apply the changes in #128 to fix CI failures?

```
Run actions/setup-python@v1
Error: Version 3.9 with arch x64 not found
```

**Comment by decfrr** [comment]
Thanks for the confirmation.
I have merged and imported the contents of the patch. Please confirm.

**Comment by r9y9** [comment]
umm, looks like numpy 2.0.0 is installed on CI 🤔 

```
Using cached numpy-2.0.0-cp39-cp39-macosx_14_0_arm64.whl (5.2 MB)
```

**Comment by decfrr** [comment]
I guess when we try to build this package using non version 2.0 numpy, we need to remove the cache in advance.

**Comment by decfrr** [comment]
Also, rather than specifying less than a specific version, I thought I should specify a version less than 2.0.
I don't think this is a point to be particular about since it is a primary response, but I think it is better this way, so I changed it.

**Comment by decfrr** [comment]
> I guess when we try to build this package using non version 2.0 numpy, we need to remove the cache in advance.

I too had a problem with ci downloading numpy 2.0. I will investigate.

**Comment by r9y9** [comment]
I guess I found the reason: `setup_requires` is deprecated

> https://setuptools.pypa.io/en/latest/userguide/dependency_management.html

> In previous versions of setuptools, this used to be accomplished with the setup_requires keyword but is now considered deprecated in favor of the [PEP 517](https://peps.python.org/pep-0517/) style described above. To peek into how this legacy keyword is used, consult our [guide on deprecated practice (WIP)](https://setuptools.pypa.io/en/latest/deprecated/index.html).

so, what about removing `setup_requires` and instead add the numpy requirement in `install_requires`?

**Comment by decfrr** [comment]
By using `install_requires` instead of `setup_requires`, the problem of numpy version had been solved.

Also, some test problems had occured, I fixed them.

1. `np.int` type has been deprecated
https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
2. change error type RuntimeError -> ValueError of wrong number test
3. Clarified `librosa.effects.time_stretch` argument

And I found a error of deprecated field `np.object`, I fixed it.

**Comment by decfrr** [comment]
I see that the test fails in 3.7.
I'd like you to run the test on other versions just to be sure.

**Comment by r9y9** [comment]
Testing other versions/platforms with github actions (https://github.com/r9y9/nnmnkwii/pull/128) and it is working OK too. I'll merge this PR after all CI gets green by either

1. dropping python 3.7 (easiest; I'm OK. It's already EOL https://devguide.python.org/versions/)
2. another idea if you prefer

**Comment by decfrr** [comment]
I have merged the changes of including dropping python 3.7 from ci.
It seems that all CI will complete jobs successfully.

**Review by r9y9 — APPROVED** [review]
LGTM, thanks!

File: tests/data/questions-radio_dnn_416.hed
Comment (r9y9) [inline tests/data/questions-radio_dnn_416.hed:1]: Sorry, I have one last comment. Is this permission change intentional? If not, could you revert it?

Comment (r9y9) [reply tests/data/questions-radio_dnn_416.hed:1]: never mind, this is my bad in #128... I will fix this after merging this PR

**Comment by decfrr** [comment]
Thank you for valuable opportunity.
I appereciate your ongoing maintenance.

**Pull request merged**
